buyer-api/src/utils/routes.js

```diff
@@ -1,46 +1,52 @@
 import web3 from './web3';
 import logger from './logger';
 
-const asyncError = fn => (req, res, next) => Promise.resolve(fn(req, res, next)).catch(next);
+// eslint-disable-next-line no-unused-vars
+export function errorHandler(err, req, res, next) {
+  if (err.failedValidation) {
+    res.boom.badData('Validation error', err.results);
+    logger.error(`${err.message
+    }\nerrors\n${err.results.errors.map(e => `  ${e.message}`)
+    }\nwarnings\n${err.results.warnings.map(e => `  ${e.message}`)
+    }`);
+  } else {
+    res.boom.badImplementation();
+    logger.error(err.message, err.fileName, err.lineNumber);
+  }
+}
 
-const errorHandler = (err, req, res, next) => { // eslint-disable-line no-unused-vars
-  logger.error(err.message, err.fileName, err.lineNumber);
-  res.boom.badImplementation();
-};
+export const isValidAddress = address => web3.utils.isAddress(address) && (
+  !/[A-F]/.test(address) ||
+  web3.utils.checkAddressChecksum(address)
+);
 
 /**
  * Middleware to check if an ethereum address is valid or not.
  * @param {String} paramName the field where the address is.
  * @public
  */
-const validateAddress = paramName => async (req, res, next) => {
-  const address = req.params[paramName];
-
-  if (web3.utils.isAddress(address)) {
-    return next();
-  }
-  res.status(400).send({
-    error: {
-      message: 'Invalid or malformed address',
-      field: paramName,
-    },
-  });
-  return undefined;
-};
-
-/**
- * Middleware to check if an notarization result is valid or not.
- * @public
- */
-export const validateFields = () => (req, res, next) => {
-  const notarizationResult = req.body;
-
-  if (!notarizationResult.orderId || typeof notarizationResult.orderId !== 'number') {
-    res.boom.badData('Validation error', { errors: ['Field "orderId" is missing or malformed'] });
-    return undefined;
-  }
-
-  return next();
-};
-
-export { errorHandler, asyncError, validateAddress };
+export function validateAddress(fieldPath) {
+  const fieldProps = typeof fieldPath === 'string' ? fieldPath.split('.') : fieldPath;
+  return async (req, res, next) => {
+    const address = fieldProps.reduce((r, c) => r && r[c], req);
+    const fieldIsValid = (
+      typeof address === 'string' && isValidAddress(address)
+    ) || address.every(isValidAddress);
```

Comment:
If you send a string that's not a `isValidAddress`, it will try to check `address.every(isValidAddress);`, but since it's a string and not an array it will throw a `500` error with `address.every is not a function`